pingcap/tiup
# cluster: support using provided topo file besides editing it while calling edit-config, add show-config command

### What problem does this PR solve? 

For subcommand `tiup cluster edit-config`, add a flag `--topology-file` to allow the user to provide a new topology file to substitute the original one besides editing it.

### What is changed and how it works?

 - For command `edit-config`, add a flag `--topology-file` to allow the user to provide a new topology file to substitute the original one besides editing it.
 - add `show-config` command


### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 1. Build the new `tiup-cluster` binary and use it as the new component in TiUP central machine.
 2. If the flag `--topology-file` is not specified, it will act like the old flow: 
     * Write Topology to a temporary file. 
     * Open file in editor.
     * Check and update Topology.
     * Save meta file.
 3. Otherwise, it will check the existence of the topology file specified by the flag `--topology-file` and will only 'Check and update Topology.' and 'Save meta file.'.
 4. While executing `./tiup-cluster show-config <cluster-name>`, the cluster's config, which is same as the one shown by opening the topology file to edit while executing `./tiup-cluster edit-config <cluster-name>`, will be printed in console.


Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation


Release notes:

```release-note
* Support using a new topology file to substitute the original one besides editing it while calling `tiup cluster edit-config`
* Add the feature `show-config` of component `tiup-cluster` to show cluster's configuration
```
